nil pointer on CSP check proof

```
2023-06-15T21:25:32.973Z DBG middleware/logger.go:158 > "GET https://api-dev.vocdoni.net/v2/elections/c5d2460186f7a7d4892df915c049f82728627302e84f525942eb030800000000 HTTP/2.0" from 181.208.114.90:43138 - 200 808B in 8.02123ms
2023-06-15T21:25:33.182Z DBG apirest/apirest.go:186 > request: {"txPayload":"ClkKVwog48MOKZ2/n6h9/JRpZwNYPGmdXg92tD+7BQLT8bVBlu8SIMXSRgGG96fUiS35FcBJ+CcoYnMC6E9SWULrAwgAAAAAGgIqACINeyJ2b3RlcyI6WzBdfRJBobKWgx3dVK+MbUVVE1VORLrBLfnrpUANxs7qQKrcdCdSRuJsOH6BCes6mTJXkb3UnV4Cp1GFgmEx5PeuKKINmwA="}
2023-06-15T21:25:33.184Z DBG transaction/vote_tx.go:173 > new vote address=0x375C715C3d47b93Da3DD4BAe7aDFa562917A9163 electionID=c5d2460186f7a7d4892df915c049f82728627302e84f525942eb030800000000 height=72352 nullifier=ed6d822a421b6ca14c80cd8b189dcde22462a593f1ee3787445f1d2754140a13 type=signature
2023-06-15T21:25:33.184Z DBG transaction/proof.go:43 > verify proof address=375c715c3d47b93da3dd4bae7adfa562917a9163 censusOrigin=3 censusRoot= electionID=c5d2460186f7a7d4892df915c049f82728627302e84f525942eb030800000000 voterID=0103f3b026cc078092e04cc8925f9fa274248fa8e057ae0eb2024b7ab847ef731887
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x2d5a9e2]

goroutine 1592436353 [running]:
go.vocdoni.io/dvote/vochain/transaction.VerifyProofOffChainCSP(0xa?, 0xc006c22100, 0x580?, {0x0, 0x0, 0x0}, {0xc008597200, 0x20, 0x20}, {0xc00d5ac8d0, ...})
	go.vocdoni.io/dvote/vochain/transaction/proof.go:150 +0xc2
go.vocdoni.io/dvote/vochain/transaction.VerifyProof(0xc0072f4e00, 0xc006c22100, 0x41?, {0x0, 0x0, 0x0}, {0xc008597200?, 0x17?, 0x0?}, {0xc00d5ac8d0, ...})
	go.vocdoni.io/dvote/vochain/transaction/proof.go:67 +0x44d
go.vocdoni.io/dvote/vochain/transaction.(*TransactionHandler).VoteTxCheck(0xc0020ab500, 0xc0060d2f50, 0x0)
	go.vocdoni.io/dvote/vochain/transaction/vote_tx.go:181 +0x1068
go.vocdoni.io/dvote/vochain/transaction.(*TransactionHandler).CheckTx(0xc0020ab500, 0xc0060d2f50, 0x0)
	go.vocdoni.io/dvote/vochain/transaction/transaction.go:81 +0xde
go.vocdoni.io/dvote/vochain.(*BaseApplication).CheckTx(0xc000190a80, {{0xc00fd44460?, 0xc0012739b0?, 0x11bc9aa?}, 0x11ec800?})
	go.vocdoni.io/dvote/vochain/app.go:282 +0x149
github.com/cometbft/cometbft/abci/client.(*localClient).CheckTxAsync(0xc0063b0420, {{0xc00fd44460?, 0x42aa4811974576e8?, 0xc00f251590?}, 0xfd44460?})
	github.com/cometbft/cometbft@v0.37.1/abci/client/local_client.go:93 +0x118
github.com/cometbft/cometbft/proxy.(*appConnMempool).CheckTxAsync(0xc0063a7818, {{0xc00fd44460?, 0x269b49d?, 0xc007a98600?}, 0x1273b68?})
	github.com/cometbft/cometbft@v0.37.1/proxy/app_conn.go:154 +0x108
github.com/cometbft/cometbft/mempool/v0.(*CListMempool).CheckTx(0xc001239d40, {0xc00fd44460, 0x9e, 0x9f}, 0xc001273cb8?, {0xe231?, {0x0?, 0xc006c22000?}})
	github.com/cometbft/cometbft@v0.37.1/mempool/v0/clist_mempool.go:254 +0x30d
github.com/cometbft/cometbft/rpc/core.BroadcastTxSync(0xc0007414a0, {0xc00fd44460, 0x9e, 0x9f})
	github.com/cometbft/cometbft@v0.37.1/rpc/core/mempool.go:35 +0xf6
github.com/cometbft/cometbft/rpc/client/local.(*Local).BroadcastTxSync(...)
	github.com/cometbft/cometbft@v0.37.1/rpc/client/local/local.go:99
go.vocdoni.io/dvote/vochain.(*BaseApplication).SetDefaultMethods.func4({0xc00fd44460?, 0x9e, 0xc001273de8?})
	go.vocdoni.io/dvote/vochain/appsetup.go:71 +0x53
go.vocdoni.io/dvote/vochain.(*BaseApplication).SendTx(0xc00fd44460?, {0xc00fd44460?, 0x9f?, 0x324ca00?})
	go.vocdoni.io/dvote/vochain/app.go:400 +0x33
go.vocdoni.io/dvote/api.(*API).submitVoteHandler(0xc0049a10a0, 0xc00f251500, 0x3e1cdc0?)
	go.vocdoni.io/dvote/api/vote.go:67 +0xe9
go.vocdoni.io/dvote/httprouter/apirest.(*API).RegisterMethod.func1({{0x3101020, 0xc00f251500}, {0xc11afe534ae92c78, 0x654fcc295484, 0x55a8f60}, {0xc00f251540, 0x2, 0x2}, 0xc0011ec7e0})
	go.vocdoni.io/dvote/httprouter/apirest/apirest.go:209 +0x74
created by go.vocdoni.io/dvote/httprouter.(*HTTProuter).routerHandler.func1
	go.vocdoni.io/dvote/httprouter/httprouter.go:261 +0x3f2
vocdoni version "5a1c401-dirty"
```

--- a/api/census_test.go
+++ b/api/census_test.go
@@ -231,9 +231,6 @@ func TestCensusProof(t *testing.T) {
 				},
 			},
 		},
-		models.CensusOrigin_OFF_CHAIN_TREE_WEIGHTED,
-		censusData.CensusID,
-		electionID,
 		state.NewVoterID(state.VoterIDTypeECDSA, key.PublicKey()),
 	)
 	qt.Assert(t, err, qt.IsNil)

--- a/vochain/transaction/election_tx.go
+++ b/vochain/transaction/election_tx.go
@@ -260,12 +260,7 @@ func (t *TransactionHandler) RegisterKeyTxCheck(vtx *vochaintx.Tx, forCommit boo
 	voterID := vstate.NewVoterID(vstate.VoterIDTypeECDSA, pubKey)
 	var valid bool
 	var weight *big.Int
-	valid, weight, err = VerifyProof(process, tx.Proof,
-		process.CensusOrigin,
-		process.CensusRoot,
-		process.ProcessId,
-		voterID,
-	)
+	valid, weight, err = VerifyProof(process, tx.Proof, voterID)
 	if err != nil {
 		return fmt.Errorf("proof not valid: %w", err)
 	}

--- a/vochain/transaction/proof.go
+++ b/vochain/transaction/proof.go
@@ -37,15 +37,19 @@ type VerifyProofFunc func(process *models.Process, proof *models.Proof,
 
 // VerifyProof is a wrapper over all VerifyProofFunc(s) available which uses the process.CensusOrigin
 // to execute the correct verification function.
-func VerifyProof(process *models.Process, proof *models.Proof,
-	censusOrigin models.CensusOrigin,
-	censusRoot, processID []byte, vID state.VoterID) (bool, *big.Int, error) {
+func VerifyProof(process *models.Process, proof *models.Proof, vID state.VoterID) (bool, *big.Int, error) {
+	if proof == nil {
+		return false, nil, fmt.Errorf("proof is nil")
+	}
+	if process == nil {
+		return false, nil, fmt.Errorf("process is nil")
+	}
 	log.Debugw("verify proof",
-		"censusOrigin", censusOrigin,
-		"electionID", fmt.Sprintf("%x", processID),
+		"censusOrigin", process.CensusOrigin,
+		"electionID", fmt.Sprintf("%x", process.ProcessId),
 		"voterID", fmt.Sprintf("%x", vID),
 		"address", fmt.Sprintf("%x", vID.Address()),
-		"censusRoot", fmt.Sprintf("%x", censusRoot),
+		"censusRoot", fmt.Sprintf("%x", process.CensusRoot),
 	)
 	// check census origin and compute vote digest identifier
 	var verifyProof VerifyProofFunc
@@ -144,8 +148,8 @@ func VerifyProofOffChainCSP(process *models.Process, proof *models.Proof,
 	key := vID.Address()
 
 	p := proof.GetCa()
-	if p == nil {
-		return false, nil, fmt.Errorf("invalid CA proof")
+	if p == nil || p.Bundle == nil {
+		return false, nil, fmt.Errorf("CSP proof or bundle are nil")
 	}
 	if !bytes.Equal(p.Bundle.Address, key) {
 		return false, nil, fmt.Errorf(
@@ -186,7 +190,7 @@ func VerifyProofOffChainCSP(process *models.Process, proof *models.Proof,
 			}
 		}
 		if !bytes.Equal(bundlePub, censusRoot) {
-			return false, nil, fmt.Errorf("csp bundle signature does not match")
+			return false, nil, fmt.Errorf("CSP bundle signature does not match")
 		}
 	case models.ProofCA_ECDSA_BLIND, models.ProofCA_ECDSA_BLIND_PIDSALTED:
 		// Blind CSP check
@@ -219,7 +223,7 @@ func VerifyProofOffChainCSP(process *models.Process, proof *models.Proof,
 				processID, rootPub.Bytes(), cspbundleDec.ProcessId, cspbundleDec.Address, signature.Bytes())
 		}
 	default:
-		return false, nil, fmt.Errorf("csp proof %s type not supported", p.Type.String())
+		return false, nil, fmt.Errorf("CSP proof %s type not supported", p.Type.String())
 	}
 	return true, bigOne, nil
 }

--- a/vochain/transaction/vote_tx.go
+++ b/vochain/transaction/vote_tx.go
@@ -178,9 +178,7 @@ func (t *TransactionHandler) VoteTxCheck(vtx *vochaintx.Tx, forCommit bool) (*vs
 			"height", height)
 
 		// Verify the proof
-		valid, weight, err := VerifyProof(process, voteEnvelope.Proof,
-			process.CensusOrigin, process.CensusRoot, process.ProcessId,
-			vote.VoterID)
+		valid, weight, err := VerifyProof(process, voteEnvelope.Proof, vote.VoterID)
 		if err != nil {
 			return nil, err
 		}

